src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -170,6 +170,12 @@ public function remove($files)
                     throw new IOException(sprintf('Failed to remove symlink "%s": ', $file).self::$lastError);
                 }
             } elseif (is_dir($file)) {
+                // Rename directory before removing it to mitigate the chance of
+                // another process writing to it during the operation.
+                // @see https://github.com/symfony/symfony/issues/27578
+                $orig_file = $file;
+                $file = '.symfony-tmp.' . substr(sha1(rand()), 0, 7);
```

Comment:
the temporary dir has to be in the same parent directory

also, we don't want to rename when removing nested folders of an already renamed one

about the tmp name, we should not use rand nor sha1 imho. Using `strrev(strtr(base64_encode(random_bytes(2)), '/=', '-.'))` would look better to me. We would delete the previously existing folder in case of collision, to have some very basic last-resort garbage collection...